torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -157,7 +158,16 @@ def dequantize(self, output_dtype=None):
 
     @staticmethod
     def _quantized_linear_op(input_tensor, weight_tensor, bias):
-        return _quantized_linear_op(input_tensor, weight_tensor, bias)
+        cls = AffineQuantizedTensor
+        func = torch.nn.functional.linear
+        args = (input_tensor, weight_tensor, bias)
+        kwargs = {}
+        if hasattr(cls, "_DISPATCH_TABLE_WITH_CONDITION"):
```

Comment:
This feels like aqt expanding on its responsibility with some APIs whose need seems to have popped up now. I do feel AffineQuantizedTensor should not take on responsibility of dispatch at all, or if it does, it should just be dq-op-q. Rest should be by derived class.

I guess here you are asking users to register their favorite dispatch but I feel we are just implementing dispatcher now. I suspect subclass approach might be cleaner

Reply:
AQT have to do dispatch because cpu/cuda or other devices code path should be optimized by default I think

Reply:
but thinking about this dispatch table, it does feel very complicated. I'm planning to revert his part of the change, and just have this as a change to refactor `_quantized_op` function for now, we can discuss later about the support for lower bit kernels

Reply:
> AQT have to do dispatch because cpu/cuda or other devices code path should be optimized by default I think

That is exactly what expanding responsibility is honestly. And this is why we end up with trying to figure out how to extend to accommodate different dispatch needs within AQT itself.